File: StrategyPattern/Bank/C++/MinusAccount.cpp
#include "MinusAccount.h"

void MinusAccount::setCredit(char newChar)
{
	Creditaraing = newChar;
	((MinusWithdrawal*) transaction)->setCredit(Creditaraing);

}

void MinusAccount::display()
{
	cout << "This is Minus Account " <<endl;
	cout << "Balance : " << balance <<endl;
	cout << "Account openning : " << openningDate.tm_year+ 1900 << "/" <<openningDate.tm_mon <<"/" <<
		openningDate.tm_mday <<" " << openningDate.tm_hour << ":" << openningDate.tm_min <<":"<< openningDate.tm_sec<< endl;
	cout << "Credit : " << Creditaraing <<endl;
}

File: StrategyPattern/Bank/JAVA/MinusAccount.java
package ACCOUNT;

public class MinusAccount  extends Account{

	char Creditrating;

	public MinusAccount(char credit)
	{
		Creditrating = credit;
		transaction = new MinusWithdrawal(Creditrating, balance);
	}
	public void setCreditraing(char credit)
	{
		Creditrating = credit;
		((MinusWithdrawal) transaction).setCredit(Creditrating);
	}
	
	@Override
	void display() {
		System.out.println("This is Minus Account");
		System.out.println("Balance  : " + balance);
		System.out.println("Account openning : " + openningDate.toString());
		System.out.println("Credit : " + Creditrating);
	}
	
	
}
